### ISSUE TITLE
Implement loading of common model formats

### ISSUE THREAD

**Issue by zgoethel12** [team-jgloom/jgloom#40]
We plan to support the following formats

Model formats 
------------- 
- [ ] MD5 - Animations and all
- [ ] OBJ 
- [ ] FBX 
- [ ] Collada - We intend to use [JCollada](http://javacollada.sourceforge.net/)
- [ ] Blend - In the very far future, as a tech demo (it would be very hard) 

For each format, comment if you are implementing it.

**Comment by zgoethel12** [comment]
I am implementing the OBJ loader and other classes.

**Comment by ecumene-software** [comment]
I can probably merge this issue into #38, and I appreciate your OBJ format contribution but we may have to adapt it to that format into the new model structure

**Comment by ecumene-software** [comment]
Thanks for the OBJ contribution, but we should lay off the file loading without having some sort of core loading format and all... See #38

**Comment by ecumene-software** [comment]
Also what'd the difference between `feat/meshes` and `feat/models`?

**Comment by zgoethel12** [comment]
`feat/meshes` for designing rendering system and mesh systems, `feat/models` is for parsing and loading code for specific model formats.

**Comment by ecumene-software** [comment]
We should probably keep those in the same branch, so we can work on the file loading and mesh system at the same time. I'll make a new branch and merge meshes to models and models to the new one?

**Comment by zgoethel12** [comment]
Go ahead.

**Comment by zgoethel12** [comment]
I think that model loading isn't something we need to focus our resources on until later.  #38 seems like a better use of our time (on this front) for now.

**Comment by ecumene-software** [comment]
Yeah, I agree. It seems like a big part of the library but there are much bigger issues.

**Comment by ecumene-software** [comment]
I'm merging `feat/meshes` into `dev` for now, and deleting the old branch. Too many changes before we actually get to the intended feature.
